cmd/internal/find/primary/meta/arrayPredicate.go

```diff
@@ -0,0 +1,151 @@
+package meta
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/puppetlabs/wash/cmd/internal/find/primary/errz"
+)
+
+// ArrayPredicate => EmptyPredicate      |
+//                   ‘[' ? ‘]’ Predicate |
+//                   ‘[' * ‘]’ Predicate |
+//                   ‘[' N ‘]’ Predicate |
+func parseArrayPredicate(tokens []string) (predicate, []string, error) {
+	if p, tokens, err := parseEmptyPredicate(tokens); err == nil {
+		return p, tokens, err
+	}
+	// EmptyPredicate did not match
+	if len(tokens) == 0 {
+		return nil, nil, errz.NewMatchError("expected an opening '['")
+	}
+	rawToken := tokens[0]
+	// For simplicity, we require rawToken to be complete. Thus, if
+	// tokens is something like ["[", "]"] or ["[", "*", "]"], then
+	// parseArrayPredicateType will return an error.
+	ptype, token, err := parseArrayPredicateType(rawToken)
+	if err != nil {
+		return nil, nil, err
+	}
+	if len(token) == 0 {
+		tokens = tokens[1:]
+	} else {
+		// token may be part of a key sequence (e.g. something like
+		// [].key2 or [][])
```

Comment:
Lots of examples of `[]` still around in code comments. Worth updating them to `[?]`?

Reply:
Updated.